src/uno.ts

```diff
@@ -0,0 +1,131 @@
+    unoConfig.theme = unoConfig.theme || {}
+    unoConfig.theme.extend = unoConfig.theme.extend || {}
+    unoConfig.theme.extend.colors =
+      unoConfig.theme.extend.colors || {}
+
+    const globalColors: any = {
+      ...(unoConfig.theme.colors || defaultColors),
+      ...unoConfig.theme.extend?.colors
+    }
+
+    // @ts-ignore
+    globalColors.primary = unoConfig.theme.extend.colors.primary = {
+      50: 'rgb(var(--color-primary-50) / <alpha-value>)',
+      100: 'rgb(var(--color-primary-100) / <alpha-value>)',
+      200: 'rgb(var(--color-primary-200) / <alpha-value>)',
+      300: 'rgb(var(--color-primary-300) / <alpha-value>)',
+      400: 'rgb(var(--color-primary-400) / <alpha-value>)',
+      500: 'rgb(var(--color-primary-500) / <alpha-value>)',
+      600: 'rgb(var(--color-primary-600) / <alpha-value>)',
+      700: 'rgb(var(--color-primary-700) / <alpha-value>)',
+      800: 'rgb(var(--color-primary-800) / <alpha-value>)',
+      900: 'rgb(var(--color-primary-900) / <alpha-value>)',
+      950: 'rgb(var(--color-primary-950) / <alpha-value>)',
+      DEFAULT: 'rgb(var(--color-primary-DEFAULT) / <alpha-value>)'
+    }
+
+    if (globalColors.gray) {
+      // @ts-ignore
+      globalColors.cool = unoConfig.theme.extend.colors.cool =
+        defaultColors.gray
+    }
+
+    // @ts-ignore
+    globalColors.gray = unoConfig.theme.extend.colors.gray = {
+      50: 'rgb(var(--color-gray-50) / <alpha-value>)',
+      100: 'rgb(var(--color-gray-100) / <alpha-value>)',
+      200: 'rgb(var(--color-gray-200) / <alpha-value>)',
+      300: 'rgb(var(--color-gray-300) / <alpha-value>)',
+      400: 'rgb(var(--color-gray-400) / <alpha-value>)',
+      500: 'rgb(var(--color-gray-500) / <alpha-value>)',
+      600: 'rgb(var(--color-gray-600) / <alpha-value>)',
+      700: 'rgb(var(--color-gray-700) / <alpha-value>)',
+      800: 'rgb(var(--color-gray-800) / <alpha-value>)',
+      900: 'rgb(var(--color-gray-900) / <alpha-value>)',
+      950: 'rgb(var(--color-gray-950) / <alpha-value>)'
+    }
+
+    const colors = excludeColors(globalColors)
+
+    // @ts-ignore
+    nuxt.options.appConfig.ui = {
+      primary: 'green',
+      gray: 'cool',
+      colors,
+      strategy: 'merge'
+    }
+
+    unoConfig.safelist = unoConfig.safelist || []
+    unoConfig.safelist.push(
+      ...generateSafelist(moduleOptions.safelistColors || [], colors)
+    )
+  })
+
+  // 2. add config template
+  const configTemplate = addTemplate({
+    filename: 'nuxtui-unocss.config.cjs',
+    write: true,
+    getContents: () => `
+      const { defineConfig } = require('unocss')
+
+      const { defaultExtractor: createDefaultExtractor } = require('tailwindcss/lib/lib/defaultExtractor.js')
+      const { customSafelistExtractor } = require(${JSON.stringify(resolve(runtimeDir, 'utils', 'colors'))})
+
+      const defaultExtractor = createDefaultExtractor({ tailwindConfig: { separator: ':' } })
+
+      module.exports = defineConfig({
```

Comment:
will need to enable [`preset-wind`](https://unocss.dev/presets/wind#installation)